source/detail/cryptography/xlsx_crypto_consumer.cpp

```
@@ -130,13 +130,13 @@ encryption_info::standard_encryption_info read_standard_encryption_info(std::ist
     }
     else
     {
-        throw xlnt::exception("invalid cipher algorithm");
+        throw xlnt::exception("invalid cipher algorithm " + std::to_string(alg_id));
```

These should be probably `invalid_file` too?

---

This is an interesting question that I also thought about. My thoughts: I would use specific exception types for exceptions that are meant to be caught (in which case we **should** definitely document them as well), while using the generic `xlnt::exception` for errors that are usually not meant to happen (things like internal parsing errors that should usually not happen). In general, if possible, I would like to reduce the number of generic `xlnt::exception`, unless the exceptions are not necessarily meant to be caught.

Now, I agree that at least some of these errors might be worth catching in some situations - which is why I slightly changed and/or documented some of them. However, I was not sure whether an `invalid_file` is correct in some cases, e.g. where we might not support some features yet, but are otherwise allowed by OOXML. In that case, we would not want to throw `invalid_file`, but e.g. `unsupported`.

However, these are things I would need to research a bit more first. Please keep this conversation open for now and I'll check it out. I'll check out several of our exceptions, including cryptography-related and compression-related exceptions.